pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -132,25 +134,35 @@ private synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableNam
         return;
       }
 
-      switch (TableNameBuilder.getTableTypeFromTableName(tableName)) {
-        case OFFLINE:
-          if (_localDiskDir != null) {
-            File fileToDelete = new File(new File(_localDiskDir, tableName), segmentId);
-            if (fileToDelete.exists()) {
-              FileUtils.deleteQuietly(fileToDelete);
-              LOGGER.info("Delete segment : " + segmentId + " from local directory : " + fileToDelete.getAbsolutePath());
-            } else {
-              LOGGER.warn("Not found local segment file for segment : " + segmentId);
-            }
-          } else {
-            LOGGER.info("localDiskDir is not configured, won't delete anything from disk");
+      final String rawTableName = TableNameBuilder.extractRawTableName(tableName);
+      if (_localDiskDir != null) {
+        File fileToMove = new File(new File(_localDiskDir, rawTableName), segmentId);
+        if (fileToMove.exists()) {
+          File deletedDir = new File(_localDiskDir, DELETED_SEGMENTS);
+          try {
+            FileUtils.moveFileToDirectory(fileToMove, deletedDir, true);
+            LOGGER.info("Moved segment {} from {} to {}", segmentId, fileToMove.getAbsolutePath(), deletedDir.getAbsolutePath());
+          } catch (IOException e) {
+            LOGGER.warn("Could not move segment {} from {} to {}", segmentId, fileToMove.getAbsolutePath(),
+                deletedDir.getAbsolutePath(), e);
           }
-          break;
-        case REALTIME:
-          LOGGER.info("No local segment file for RealtimeSegment in Controller");
-          break;
-        default:
-          throw new UnsupportedOperationException("Not support ResourceType for semgnet - " + segmentId);
+        } else {
+          CommonConstants.Helix.TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableName);
+          switch (tableType) {
+            case OFFLINE:
+              LOGGER.warn("Not found local segment file for segment {}" + fileToMove.getAbsolutePath());
+              break;
+            case REALTIME:
+              if (SegmentName.isLowLevelConsumerSegmentName(segmentId)) {
+                LOGGER.warn("Not found local segment file for segment {}" + fileToMove.getAbsolutePath());
+              }
+              break;
+            default:
+              LOGGER.warn("Unsupported table type {} when deleting segment {}", tableType, segmentId);
+          }
+        }
+      } else {
+        LOGGER.info("localDiskDir is not configured, won't delete segment {} from disk", segmentId);
```

Comment:
This shouldn't happen. Constructor should throw an error if localDiskDir is not configured.